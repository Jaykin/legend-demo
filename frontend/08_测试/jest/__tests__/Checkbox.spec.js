/**
 * Created by Administrator on 2017/2/6.
 */
import React from 'react';
import Checkbox from '../src/components/Checkbox';
import renderer from 'react-test-renderer';

import {shallow} from 'enzyme';

    // react-test-renderer
test('Checkbox changes the text after click', () => {
    const component = renderer.create(
        <Checkbox labelOn="On" labelOff="Off" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.children[0].props.onChange();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


    // enzyme
test('CheckboxWithLabel changes the text after click', () => {
    // Render a checkbox with label in the document
    const checkbox = shallow(
        <Checkbox labelOn="On" labelOff="Off" />
    );

    expect(checkbox.text()).toEqual('Off');

    checkbox.find('input').simulate('change');

    expect(checkbox.text()).toEqual('On');
});
