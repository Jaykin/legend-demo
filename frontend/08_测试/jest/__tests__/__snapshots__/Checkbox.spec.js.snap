exports[`test Checkbox changes the text after click 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox" />
  Off
</label>
`;

exports[`test Checkbox changes the text after click 2`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox" />
  On
</label>
`;
